pulsar-broker/src/main/java/com/yahoo/pulsar/broker/namespace/OwnedBundle.java

```diff
@@ -101,16 +101,29 @@ public void handleUnloadRequest(PulsarService pulsar) throws Exception {
             this.nsLock.writeLock().unlock();
         }
 
-        int unloadedTopics = 0;
+        int unloadedTopics = -1;
         try {
             LOG.info("Disabling ownership: {}", this.bundle);
-            pulsar.getNamespaceService().getOwnershipCache().disableOwnership(this.bundle);
-
-            // Handle unload of persistent topics
-            unloadedTopics = pulsar.getBrokerService().unloadServiceUnit(bundle).get();
-            pulsar.getNamespaceService().getOwnershipCache().removeOwnership(bundle);
+            pulsar.getNamespaceService().getOwnershipCache().updateBundleState(this.bundle, false);
+
+            // close topics forcefully
+            try {
+                unloadedTopics = pulsar.getBrokerService().unloadServiceUnit(bundle).get();
+            } catch (Exception e) {
+                // ignore topic-close failure to unload bundle
```

Comment:
I was saying to verify that ML.close() will remove the ML from the factory even in the presence of errors. Errors could happen when closing the cursor or closing the last data ledger.